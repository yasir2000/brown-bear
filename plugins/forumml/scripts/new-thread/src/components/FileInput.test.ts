/*
 * Copyright (c) Enalean, 2021 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import FileInput from "./FileInput.vue";
import { createNewThreadLocalVue } from "../helpers/local-vue-for-test";
import Vue from "vue";

describe("FileInput", () => {
    it("Displays the input field to enter a file", async () => {
        const wrapper = shallowMount(FileInput, {
            localVue: await createNewThreadLocalVue(),
            propsData: {
                index: 123,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("Asks to parent component to remove the file if user clicks on delete button", async () => {
        const wrapper = shallowMount(FileInput, {
            localVue: await createNewThreadLocalVue(),
            propsData: {
                index: 123,
            },
        });

        const button = wrapper.find('[data-test="delete-file"]');
        button.trigger("click");

        await Vue.nextTick();

        expect(wrapper.emitted().removeFile).toBeTruthy();
    });
});
