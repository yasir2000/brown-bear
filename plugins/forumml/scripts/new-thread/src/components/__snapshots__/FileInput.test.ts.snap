// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput Displays the input field to enter a file 1`] = `
<div
  class="tlp-form-element tlp-form-element-append forumml-post-new-thread-file-component"
>
  <span
    class="tlp-button-secondary forumml-post-new-thread-file-button"
  >
    <i
      class="fa fa-upload tlp-button-icon"
    />
     
    <translate-stub
      tag="span"
    >
      Browse...
    </translate-stub>
     
    <input
      class="tlp-input forumml-post-new-thread-file-input"
      id="forumml-file-123"
      name="files[]"
      required="required"
      type="file"
    />
  </span>
   
  <button
    class="tlp-append tlp-button-primary tlp-button-outline"
    data-test="delete-file"
    type="button"
  >
    <i
      aria-hidden="true"
      class="fas fa-trash"
    />
  </button>
</div>
`;
